CosmosApiChangeFeedForMessageRetention/__test__/index.test.ts

```diff
@@ -0,0 +1,123 @@
+import * as O from "fp-ts/lib/Option";
+import * as TE from "fp-ts/lib/TaskEither";
+import * as E from "fp-ts/lib/Either";
+
+import { Ttl } from "@pagopa/io-functions-commons/dist/src/utils/cosmosdb_model_ttl";
+import { RejectedMessageStatusValueEnum } from "@pagopa/io-functions-commons/dist/generated/definitions/RejectedMessageStatusValue";
+import { Context } from "@azure/functions";
+import { handleSetTTL, isEligibleForTTL, RELEASE_TIMESTAMP } from "../handler";
+import {
+  aMessageStatus,
+  mockMessageModel,
+  mockMessageStatusModel,
+  mockPatch,
+  mockUpdateTTLForAllVersions
+} from "../../__mocks__/message";
+
+import { mockProfileFindLast, mockProfileModel } from "../../__mocks__/profile";
+
+const ttl = 500 as Ttl;
+
+const anEligibleDocument = {
+  ...aMessageStatus,
+  status: RejectedMessageStatusValueEnum.REJECTED
+};
+
+const mockContext = ({
+  bindings: {},
+  done: jest.fn(),
+  log: jest.fn()
+} as unknown) as Context;
+
+const mockDocuments = [
+  anEligibleDocument,
+  anEligibleDocument,
+  anEligibleDocument,
+  aMessageStatus,
+  anEligibleDocument,
+  aMessageStatus
+];
+
+describe("isEligibleForTTL", () => {
+  it("should return a string if the status is not REJECTED", async () => {
+    const r = await isEligibleForTTL(aMessageStatus)();
+    expect(E.isLeft(r)).toBeTruthy();
+    if (E.isLeft(r)) {
+      expect(r.left).toBe("This message status is not rejected");
+    }
+  });
+
+  it("should return a string if the _ts is after the RELEASE_TIMESTAMP", async () => {
+    const r = await isEligibleForTTL({
+      ...aMessageStatus,
+      _ts: 2670524345,
+      status: RejectedMessageStatusValueEnum.REJECTED
+    })();
+    expect(E.isLeft(r)).toBeTruthy();
+    if (E.isLeft(r)) {
+      expect(r.left).toBe(
+        `the timestamp of the document ${
+          aMessageStatus.id
+        } (${2670524345}) is after the RELEASE_TIMESTAMP ${RELEASE_TIMESTAMP}`
+      );
+    }
+  });
+
+  it("should return a string if the document already has a ttl", async () => {
+    const r = await isEligibleForTTL({
+      ...aMessageStatus,
+      status: RejectedMessageStatusValueEnum.REJECTED,
+      ttl
+    })();
+    expect(E.isLeft(r)).toBeTruthy();
+    if (E.isLeft(r)) {
+      expect(r.left).toBe(
+        `the document ${aMessageStatus.id} has a ttl already`
+      );
+    }
+  });
+
+  it("should return the retrieved document if it is eligible", async () => {
+    const r = await isEligibleForTTL(anEligibleDocument)();
+    expect(E.isRight(r)).toBeTruthy();
+    if (E.isRight(r)) {
+      expect(r.right).toBe(anEligibleDocument);
+    }
+  });
+});
+
+describe("handleSetTTL", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.restoreAllMocks();
+  });
+
+  it("should call the findLastVersionByModelId 3 times but never set the ttl", async () => {
```

Comment:
It says 3, but it checks for 4. Also, we should explain something about conditions: what type of message status we have as input?

Reply:
Sorry I forgot to update the string passed to the `it`, that 3 should be a 4 cause we are passing 4 eligible documents here, I'll add a little comment to explain the scenario as well 😃